--- a/vfs0090.h
+++ b/vfs0090.h
@@ -343,10 +343,11 @@ static const VfsDataExchange INIT_SEQUENCES[] = {
     .rsp_length = 68,
   },
   {
+    .weak_match = TRUE,
     .msg = INIT_SEQUENCE_MSG3,
     .msg_length = G_N_ELEMENTS (INIT_SEQUENCE_MSG3),
     .rsp = NULL,
-    .rsp_length = 84,
+    .rsp_length = 108 /* or 108 */,
   },
   {
     .msg = INIT_SEQUENCE_MSG4,
@@ -737,16 +738,18 @@ static const VfsDataExchange ACTIVATE_SEQUENCES[] = {
     .rsp_length = G_N_ELEMENTS (ACTIVATE_SEQUENCE_RSP345),
   },
   {
+    .weak_match = TRUE,
     .msg = ACTIVATE_SEQUENCE_MSG67,
     .msg_length = G_N_ELEMENTS (ACTIVATE_SEQUENCE_MSG67),
     .rsp = NULL,
-    .rsp_length = 84,
+    .rsp_length = 108,
   },
   {
+    .weak_match = TRUE,
     .msg = ACTIVATE_SEQUENCE_MSG67,
     .msg_length = G_N_ELEMENTS (ACTIVATE_SEQUENCE_MSG67),
     .rsp = NULL,
-    .rsp_length = 84,
+    .rsp_length = 108,
   },
   {
     .msg = SCAN_MATRIX,
